forgerock-openbanking-analytics-core/src/main/java/com/forgerock/openbanking/analytics/upgrade/UpgradeStep_4_1_4.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.analytics.upgrade;
+
+import com.forgerock.openbanking.analytics.model.entries.*;
+import com.forgerock.openbanking.analytics.models.TokenUsageEntry;
+import com.forgerock.openbanking.upgrade.exceptions.UpgradeException;
+import com.forgerock.openbanking.upgrade.model.UpgradeMeta;
+import com.forgerock.openbanking.upgrade.model.UpgradeStep;
+import lombok.extern.slf4j.Slf4j;
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.BulkOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StopWatch;
+
+@Service
+@UpgradeMeta(version = "4.1.4")
+@Slf4j
+public class UpgradeStep_4_1_4 implements UpgradeStep {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public boolean upgrade() throws UpgradeException {
+        StopWatch elapsedTime = new StopWatch();
+        elapsedTime.start();
+        log.info("-----------------------------------------------------------------------");
+        log.debug("Start "+this.getClass().getName()+" to upgrading to version " + this.getClass().getDeclaredAnnotation(UpgradeMeta.class).version());
```

Comment:
Minor - this would read better:

```
log.debug("Start upgrading {} to version {}", this.getClass().getName(), this.getClass().getDeclaredAnnotation(UpgradeMeta.class).version());
```

Reply:
True, Old school vice.
Done.